src/Utilities/Compiler/AdditionalFileProvider.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Analyzer.Utilities
+{
+    /// <summary>
+    /// Provider that allows analyzers to easily find and use
+    /// <see href="https://github.com/dotnet/roslyn/blob/master/docs/analyzers/Using%20Additional%20Files.md">additional files</see>.
+    /// </summary>
+    internal class AdditionalFileProvider
+    {
+        private readonly ImmutableArray<AdditionalText> _additionalFiles;
+
+        internal AdditionalFileProvider(params AdditionalText[] additionalFiles)
```

Comment:
This constructor seems unused.

Reply:
It was used by AdditionalFileProviderTests that I originally forgot to include. I've now included them and just used the ImmutableArray override